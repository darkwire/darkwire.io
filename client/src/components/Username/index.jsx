import React, { Component } from 'react';
import PropTypes from 'prop-types';
import randomColor from 'randomcolor';

class Username extends Component {
  render() {
    return (
      <span className="username" style={{ color: randomColor({ seed: this.props.username, luminosity: 'light' }) }}>
        {this.props.username}
      </span>
    );
  }
}

Username.propTypes = {
  username: PropTypes.string.isRequired,
};

export default Username;
