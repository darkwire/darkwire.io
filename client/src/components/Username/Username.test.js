import React from 'react';
import { render } from '@testing-library/react';

import Username from '.';

test('Username component is displaying', async () => {
  const { asFragment } = render(<Username username="paul" />);

  expect(asFragment()).toMatchSnapshot();
});
