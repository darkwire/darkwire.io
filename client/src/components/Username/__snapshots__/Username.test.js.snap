// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Username component is displaying 1`] = `
<DocumentFragment>
  <span
    class="username"
    style="color: rgb(157, 242, 249);"
  >
    paul
  </span>
</DocumentFragment>
`;
